pretyping/coercion.ml

```diff
@@ -622,6 +622,31 @@ let lookup_reversible_path_to_common_point env sigma ~src_expected ~src_inferred
   in
     aux r
 
+type hook = env -> evar_map -> flags:Evarconv.unify_flags -> constr ->
+  inferred:types -> expected:types -> (evar_map * constr) option
+type hook_rec = { hr_id : int; hr_active : bool; hr_hook : hook }
+
+let active_hooks = ref ([] : hook list)
+
+let add_hook =
+  let all_hooks = ref ([] : hook_rec list) in
+  let next_id = ref 0 in
```

Comment:
This is not stable by process spawning, and the code is quite contrived in addition. You must use a static, unique name string registering mechanism and only store the string in the summary instead.

Reply:
Thanks. Replaced the inproperly generated int with an externally provided string. Only strings are stored in the summary.